presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -422,17 +423,9 @@ private static Throwable failureCause(StageInfo stageInfo)
         @Override
         public SqlQueryExecution createQueryExecution(QueryId queryId, String query, Session session, Statement statement)
         {
-            int initialHashPartitions;
+            int initialHashPartitions = getHashPartitionCount(session, this.initialHashPartitions);
```

Comment:
I was debating which way this should go, and I thought that since big_query_enabled and big query hash partitions are related properties, if big query is enabled, we'd want the initial hash partitions count for big queries. If we do it the other way round, it gives more flexibility to run big queries with different hash partitions, but setting the big query hash partition count wouldn't mean anything.